eo-maven-plugin/src/main/java/org/eolang/maven/ProbeMojo.java

```diff
@@ -0,0 +1,248 @@
+        }
+        final Collection<Tojo> tojos = this.scopedTojos().select(
+            row -> row.exists(AssembleMojo.ATTR_XMIR2)
+                && !row.exists(AssembleMojo.ATTR_PROBED)
+        );
+        final CommitHash hash = new ChResolve(
+            this.offlineHashFile, this.offlineHash, this.tag
+        );
+        if (this.objectionary == null) {
+            this.objectionary = new OyFallbackSwap(
+                new OyHome(
+                    new ChNarrow(hash),
+                    this.outputPath
+                ),
+                new OyRemote(hash),
+                this.forceUpdate()
+            );
+        }
+        final Collection<String> probed = new HashSet<>(1);
+        for (final Tojo tojo : tojos) {
+            final Path src = Paths.get(tojo.get(AssembleMojo.ATTR_XMIR2));
+            final Collection<String> names = this.probes(src);
+            int count = 0;
+            for (final String name : names) {
+                if (!this.objectionary.contains(name)) {
+                    continue;
+                }
+                ++count;
+                final Tojo ftojo = this.scopedTojos().add(name);
+                if (!ftojo.exists(AssembleMojo.ATTR_VERSION)) {
+                    ftojo.set(AssembleMojo.ATTR_VERSION, "*.*.*");
+                }
+                ftojo.set(AssembleMojo.ATTR_DISCOVERED_AT, src);
+                probed.add(name);
+            }
+            tojo.set(AssembleMojo.ATTR_HASH, new ChNarrow(hash).value());
+            tojo.set(AssembleMojo.ATTR_PROBED, Integer.toString(count));
+        }
+        if (tojos.isEmpty()) {
+            if (this.scopedTojos().select(row -> true).isEmpty()) {
+                Logger.warn(this, "Nothing to probe, since there are no programs");
+            } else {
+                Logger.info(this, "Nothing to probe, all programs checked already");
+            }
+        } else if (probed.isEmpty()) {
+            Logger.info(
+                this, "No probes found in %d programs",
+                tojos.size()
+            );
+        } else {
+            Logger.info(
+                this, "Found %d probes in %d programs: %s",
+                probed.size(), tojos.size(), probed
+            );
+        }
+    }
+
+    /**
+     * Find all probes found in the provided XML file.
+     *
+     * @param file The .xmir file
+     * @return List of foreign objects found
+     * @throws FileNotFoundException If not found
+     */
+    private Collection<String> probes(final Path file)
+        throws FileNotFoundException {
+        final XML xml = new XMLDocument(file);
+        final Collection<String> probed = new TreeSet<>(
+            new ListOf<>(
+                new Filtered<>(
+                    obj -> !obj.isEmpty(),
+                    xml.xpath("//metas/meta[head/text() = 'probe']/tail/text()")
+                )
+            )
+        );
+        final Collection<String> ret = new TreeSet<>();
+        probed.forEach(
```

Comment:
@MikhailLipanin Since you are using `cactoos` approach above, it looks convenient to use the same approach here using `Filtered` and `Mapped` classes, then wrap result using `new TreeSet<>(...)`

Reply:
added todo